lib/dataset/sized_dict.cpp

```diff
@@ -379,8 +378,13 @@ core::Dict<Key, Value> union_(const SizedDict<Key, Value> &a,
   for (const auto &[key, val] : a)
     out.insert_or_assign(key, val);
   for (const auto &[key, val] : b) {
-    if (const auto it = a.find(key); it != a.end()) {
-      expect::matching_coord(it->first, it->second, val, opname);
+    if (const auto it = out.find(key); it != out.end()) {
+      if (it->second.is_aligned() == val.is_aligned())
+        expect::matching_coord(key, it->second, val, opname);
+      else if (val.is_aligned())
+        it->second = val; // aligned overrides unaligned
+      else if (!it->second.is_aligned())
+        out.erase(key); // conflicting unaligned coord
```

Comment:
Where is the check that it is conflicting? `intersection` below does an `equals_nan` check.

Reply:
Good catch